Adjust and move Effective Dart's linter mention (#4345)
dart-lang/site-www: src/_guides/language/effective-dart/index.md

are common sense and we're all reasonable people. The goal, as always, is nice,
readable and maintainable code.

> The Dart analyzer has a linter to help you write good, consistent code.
> If a linter rule exists that can help you follow a guideline,
> then the guideline links to that rule using the following format:

Reviewer: **suggestion:** Based on the other pages that you reference, I would update the language somewhat.
```suggestion
The Dart analyzer provides a linter to identify potential source code problems.
If your code triggers a [linter rule](/tools/linter-rules),
then a link to that rule appears. The link uses the following format:
```

**question:** Where does the link display? In the IDE? It's a little unclear here.

Author: This text is describing the format of the Effective Dart pages, where if that guideline has a cooresponding linter rule, then it links to it. The goal is to help readers understand what that mention is.

Here is an example:
<img width="553" alt="Effective Dart linter rule link" src="https://user-images.githubusercontent.com/18372958/200401291-217511ff-4eaf-4d0f-b15e-4071087341e7.png">

Reviewer: @parlough : Then my changes probably don't apply. Maybe make it more clear that you're writing about the highlighted issue in the documentation, not that it's in the code or IDE.

{% include linter-rule-mention.md rule="unnecessary_getters_setters" %}

> For help using the linter,
> see [Enabling linter rules][]
> and the list of [linter rules][].

Reviewer: **suggestion:** Encourage action in "see also" type passages.
```suggestion
To learn how to use the linter,
see [Enabling linter rules][]
and the list of [linter rules][].
```
